modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/internal/DiscoveryDataPacket.java

```diff
@@ -271,13 +282,16 @@ private boolean mapsEqual(Map<Integer, byte[]> m1, Map<Integer, byte[]> m2) {
             Marshaller marsh,
             ClassLoader clsLdr,
             boolean clientNode,
-            IgniteLogger log
+            IgniteLogger log,
+            boolean compressData
     ) {
         Map<Integer, Serializable> res = U.newHashMap(src.size());
 
         for (Map.Entry<Integer, byte[]> binEntry : src.entrySet()) {
             try {
-                Serializable compData = marsh.unmarshal(binEntry.getValue(), clsLdr);
+                Serializable compData = COMPRESSION_LEVEL < 0 || !compressData ?
```

Comment:
why do we need this check COMPRESSION_LEVEL < 0? Seems redundant, we will fail with exception if compression is negative.